### DISCUSSION TITLE
post/how-to-resample-audio-with-naudio

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#12] [Announcements]
# post/how-to-resample-audio-with-naudio

Mark Heath's Development Blog

https://markheath.net/post/how-to-resample-audio-with-naudio

**Comment by ericv911** [comment]
That media foundation resampler is actually a very good tool!   I haven't tried any of the others mentioned in this article yet. 
I am just beginning with adding sounds to a simple project I made and found NAudio a couple of days ago. (I must sound archaic with ChatGPT around the corner, but anyway)  
Adding sound effects naively using C# native Mediaplayer, resulted in serious memory leaks as the things just kept clogging up memory without getting disposed in time,  
NAudio was itself already extremely useful preventing that with the AudioPlayBackEngine you provided in an article somewhere. I had to feed it  constant samplerate  audiofiles though (converting all the audio files to 44.1 KHz manually with FFMpeg).   I couldn't quite figure out how to change  the samplerate of an audiostream myself in C# as I never worked with audio before. 
But now I found this media foundation resampler -> converting sound-effects to the right samplerate dynamically while  starting up the project is child's play now, even for me.  
Thanks for a very helpful library and tool!
